deepcell/metrics.py

```diff
@@ -259,22 +259,35 @@ def _calc_iou(self):
             self.seg_thresh = np.zeros((self.n_true, self.n_pred))
 
         # Use bounding boxes to find masks that are likely to overlap
-        true_bbox = regionprops_table(self.y_true)['bbox']
-        pred_bbox = regionprops_table(self.y_pred)['bbox']
+        y_true_props = skimage.measure.regionprops(np.squeeze(self.y_true.astype('int')))
+        y_true_boxes = [np.array(y_true_prop.bbox) for y_true_prop in y_true_props]
+        y_true_boxes = np.array(y_true_boxes).astype('double')
+        y_true_box_labels = [int(y_true_prop.label) for y_true_prop in y_true_props]
+        
+        y_pred_props = skimage.measure.regionprops(np.squeeze(self.y_pred.astype('int')))
+        y_pred_boxes = [np.array(y_pred_prop.bbox) for y_pred_prop in y_pred_props]
+        y_pred_boxes = np.array(y_pred_boxes).astype('double')
+        y_pred_box_labels = [int(y_pred_prop.label) for y_pred_prop in y_pred_props]
+        
+        # has the form [gt_bbox, res_bbox]
+        overlaps = compute_overlap(y_true_boxes, y_pred_boxes)
+
+        # Find the bboxes that have overlap at all (ind_ corresponds to box number - starting at 0)
+        ind_true, ind_pred = np.nonzero(overlaps)
+
+        for index in range(ind_true.shape[0]):
+
+            iou_y_true_idx = y_true_box_labels[ind_true[index]]
+            iou_y_pred_idx = y_pred_box_labels[ind_pred[index]]
+            intersection = np.logical_and(self.y_true==iou_y_true_idx, self.y_pred==iou_y_pred_idx)
+            union = np.logical_or(self.y_true==iou_y_true_idx, self.y_pred==iou_y_pred_idx)
+            # Subtract 1 from index to account for skipping 0            
+            self.iou[iou_y_true_idx-1, iou_y_pred_idx-1] = intersection.sum() / union.sum()
 
-        overlaps = compute_overlap(true_bbox, pred_bbox)
-        ind_true, ind_pred = np.nonzeros(overlaps)
-
-        for index in range(self.y_true.shape[0]):
-            intersection = np.logical_and(self.y_true == ind_true[index] + 1,
-                                          self.y_pred == ind_pred[index] + 1)
-            union = np.logical_or(self.y_true == ind_true[index] + 1,
-                                  self.y_pred == ind_pred[index] + 1)
-            self.iou[ind_true[index], ind_pred[index]] = intersection.sum() / union.sum()
 
             if (self.seg is True) & \
```

Comment:
don't need to check `if True is True`, you can just say `if True`.  This might clear up the logic here a bit.

Is the `0.5` supposed to be a parameter somewhere?

Reply:
Done.

I believe that is a set parameter from SEG metric adapted from Maska et al. mentioned in the opening docstring